maths/radix2_fft.py

```diff
@@ -174,20 +174,25 @@ def __multiply(self):
 
     # Overwrite __str__ for print(); Shows A, B and A*B
     def __str__(self):
-        A = "A = "
-        B = "B = "
-        C = "A*B = "
-        for i in range(self.len_A):
-            A += str(self.polyA[i]) + "*x^" + str(i) + " + "
-        for i in range(self.len_B):
-            B += str(self.polyB[i]) + "*x^" + str(i) + " + "
-        for i in range(self.len_B + self.len_A - 1):
-            C += (
-                str(self.product[i])
-                + "*x^"
-                + str(i)
-                + " + "
-            )

```

Comment:
A = "A = " + ".join(f"{item}*x^{i}" for i, item in enumerate(self.polyA))

Reply:
I switched it to `self.polyA[:self.len_A]` instead as `self.polyA` has some leading zeros necessary for the FFT.